openedx/features/philu_courseware/helpers.py

```diff
@@ -46,41 +43,48 @@ def get_nth_chapter_link(course, chapter_index=0):
     return base_url + course_target
 
 
-def get_competency_assessments_score(user, course_id, chapter_id):
+def get_competency_assessments_score(user, course_id, chapter_id, section_id, is_pre):
     """
     Return competency assessments scores of user in chapter of specified course
 
     :param user: user
     :param course_id: the identifier for the course.
     :param chapter_id: chapter url_name.
+    :param section_id: section url_name.
+    :param is_pre: Boolean to check if the assessment is pre or post.
     :return: assessments score dictionary
     :raises:
         NotFound: if chapter, pre or post assessment not found
     """
+
     course_key = CourseKey.from_string(course_id)
-    course = get_course_with_access(user, 'load', course_key,
-                                    check_if_enrolled=True)
-    course_grade = CourseGradeFactory().read(user, course)
-    chapter_grade = course_grade.chapter_grades.values()
-    try:
-        chapter = next(chapter for chapter in chapter_grade
-                       if chapter['url_name'] == chapter_id)
-    except StopIteration:
-        raise NotFound(_('Chapter not found'))
+    course_block = get_course_block_key(course_key)
+    course_struct = generate_course_structure(course_key)['structure']
 
     pre_assessment_score = post_assessment_score = None
-    pre_assessment_attempted = all_pre_assessment_attempted = all_post_assessment_attempted = False
-    for section in chapter['sections']:
-        if is_pre_assessment(section):
-            pre_assessment_score = section.all_total.earned
-            pre_assessment_attempted = bool(section.all_total.first_attempted)
-            all_pre_assessment_attempted = is_all_attempted(section)
-        elif is_post_assessment(section):
-            post_assessment_score = section.all_total.earned
-            all_post_assessment_attempted = is_all_attempted(section)
-
-    if pre_assessment_score is None or post_assessment_score is None:
-        raise NotFound(_('Pre or post assessment not found'))
+    pre_assessment_attempted = all_pre_assessment_attempted = all_post_assessment_attempted = None
+
+    if is_pre == 'True':
+        pre_block = BlockUsageLocator.from_string(get_complete_block_key(course_block, section_id, 'sequential'))
+        pre_assessment_score, pre_assessment_attempted, all_pre_assessment_attempted = get_assessment_score(
+            user, pre_block)
+
```

Comment:
We aren't handling this case... As discussed with @ahmed-faraz46 it will assess that module even if that assessment is missing. It won't break anything.